### PR TITLE
Extend ghc support to 8.8 and 8.10

### LINKED ISSUE #118: Support for GHC 8.8 and 8.10
TODO:

- [ ] Add support to GHC 8.8 and 8.10 ([update travis here](https://github.com/stackbuilders/dotenv-hs/blob/master/.travis.yml#L5))
- [ ] Install the latest version (`8.10`) with OS X using [ghcup](https://github.com/stackbuilders/dotenv-hs/blob/master/.travis.yml#L15)
- [ ] Use the latest version (`8.10`) to build the Linux static binary ([here](https://github.com/stackbuilders/dotenv-hs/blob/master/.travis.yml#L53))

### PR DESCRIPTION
**Pull request by TristanCacqueray**
## Extend ghc support to 8.8 and 8.10

This change updates the travis configuration to test the new version.

Fixes #118

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
The project seems to work with ghc 8.10, at least provided by the fedora ghc:8.10 module:

```
[root@5130136a3ff2 dotenv-hs]# cabal configure --enable-tests
Warning: The configure command is a part of the legacy v1 style of cabal
usage.

Please switch to using either the new project style and the new-configure
command or the legacy v1-configure alias as new-style projects will become the
default in the next version of cabal-install. Please file a bug if you cannot
replicate a working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Warning: Unknown/unsupported 'ghc' version detected (Cabal 2.4.0.1 supports
'ghc' version < 8.7): /usr/bin/ghc is version 8.10.1
Resolving dependencies...
Configuring dotenv-0.8.0.4...
[root@5130136a3ff2 dotenv-hs]# cabal test
Warning: The test command is a part of the legacy v1 style of cabal usage.

Please switch to using either the new project style and the new-test command
or the legacy v1-test alias as new-style projects will become the default in
the next version of cabal-install. Please file a bug if you cannot replicate a
working v1- use case with the new-style commands.

For more information, see: https://wiki.haskell.org/Cabal/NewBuild

Preprocessing library for dotenv-0.8.0.4..
Building library for dotenv-0.8.0.4..
Preprocessing test suite 'dotenv-test' for dotenv-0.8.0.4..
Building test suite 'dotenv-test' for dotenv-0.8.0.4..
[ 1 of 16] Compiling Configuration.Dotenv.Environment ( src/Configuration/Dotenv/Environment.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Environment.o )
[ 2 of 16] Compiling Configuration.Dotenv.ParsedVariable ( src/Configuration/Dotenv/ParsedVariable.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/ParsedVariable.o )
[ 3 of 16] Compiling Configuration.Dotenv.Parse ( src/Configuration/Dotenv/Parse.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Parse.o )
[ 4 of 16] Compiling Configuration.Dotenv.ParseSpec ( spec/Configuration/Dotenv/ParseSpec.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/ParseSpec.o )
[ 5 of 16] Compiling Configuration.Dotenv.Scheme.Types ( src/Configuration/Dotenv/Scheme/Types.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Scheme/Types.o )
[ 6 of 16] Compiling Configuration.Dotenv.Scheme.Parser ( src/Configuration/Dotenv/Scheme/Parser.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Scheme/Parser.o )
[ 7 of 16] Compiling Configuration.Dotenv.Scheme.ParseSpec ( spec/Configuration/Dotenv/Scheme/ParseSpec.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Scheme/ParseSpec.o )
[ 8 of 16] Compiling Configuration.Dotenv.Scheme.Helpers ( src/Configuration/Dotenv/Scheme/Helpers.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Scheme/Helpers.o )
[ 9 of 16] Compiling Configuration.Dotenv.Scheme ( src/Configuration/Dotenv/Scheme.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Scheme.o )
[10 of 16] Compiling Configuration.Dotenv.SchemeSpec ( spec/Configuration/Dotenv/SchemeSpec.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/SchemeSpec.o )
[11 of 16] Compiling Configuration.Dotenv.Types ( src/Configuration/Dotenv/Types.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Types.o )
[12 of 16] Compiling Configuration.Dotenv ( src/Configuration/Dotenv.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv.o )
[13 of 16] Compiling Configuration.Dotenv.Text ( src/Configuration/Dotenv/Text.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/Text.o )
[14 of 16] Compiling Configuration.Dotenv.TextSpec ( spec/Configuration/Dotenv/TextSpec.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/Dotenv/TextSpec.o )
[15 of 16] Compiling Configuration.DotenvSpec ( spec/Configuration/DotenvSpec.hs, dist/build/dotenv-test/dotenv-test-tmp/Configuration/DotenvSpec.o )
[16 of 16] Compiling Main             ( spec/Spec.hs, dist/build/dotenv-test/dotenv-test-tmp/Main.o )
Linking dist/build/dotenv-test/dotenv-test ...
Running 1 test suites...
Test suite dotenv-test: RUNNING...
Test suite dotenv-test: PASS
Test suite logged to: dist/test/dotenv-0.8.0.4-dotenv-test.log
1 of 1 test suites (1 of 1 test cases) passed.

```

**Comment by CristhianMotoche** [comment]
Thanks @TristanCacqueray for the contribution. I think we still need to test it for the OS X build. Probably, only one of them GHC 8.10 if possible.

Also, I'm not sure why the `after_install` and `before_install` directives weren't skipped in the Test stage. I'll try to fix that first.

**Comment by TristanCacqueray** [comment]
I don't have access to OS X and just tested 8.10 on Linux.

**Comment by CristhianMotoche** [comment]
Hello @TristanCacqueray The PR looks good. 
Could you please rebase your branch with `master`?

Also, could you please update the `dotenv.cabal` file [here](https://github.com/stackbuilders/dotenv-hs/blob/master/dotenv.cabal#L2) as a minor version and the [Changelog](https://github.com/stackbuilders/dotenv-hs/blob/master/CHANGELOG.md)? 

Thanks in advance.

**Comment by CristhianMotoche** [comment]
Hello @TristanCacqueray It seems there are some errors in the [CI](https://travis-ci.org/github/stackbuilders/dotenv-hs/builds/677373743). Probably a dependency isn't compatible with GHC 8.8 or 8.10 and we need to update the boundaries. Let me know if you need a hand with that.

**Comment by TristanCacqueray** [comment]
I'm not able to reproduce the failure locally, perhaps it is because i'm using `cabal-install version 3.2.0.0` and travis is `cabal-install version 2.0.0.1`. Last change tries to update cabal-install...

**Comment by TristanCacqueray** [comment]
It seems like the .travis file change is not evaluated by the job.

**Comment by CristhianMotoche** [comment]
Hi @TristanCacqueray 
I think thex change was evaluated. However, take into account that `ghcup` is used for the [mac build](https://travis-ci.org/github/stackbuilders/dotenv-hs/jobs/680309687).

**Comment by TristanCacqueray** [comment]
@CristhianMotoche looking at 77844e5 build [logs](https://travis-ci.org/github/stackbuilders/dotenv-hs/jobs/681535773), I'm not able to find the added `cabal --version` command output.

**Comment by CristhianMotoche** [comment]
Hello @TristanCacqueray I see you added those debug lines in the `.travis/ghcup.sh` script. This script is only executed for the mac build. I can see your debugging line in that build [here](https://travis-ci.org/github/stackbuilders/dotenv-hs/jobs/681535780#L799).

The [logs](https://travis-ci.org/github/stackbuilders/dotenv-hs/jobs/681535773) that you're linking belong to a linux build which uses the cabal version [2.0](https://travis-ci.org/github/stackbuilders/dotenv-hs/jobs/681535773#L236). I think it's possible to specify the cabal version for those builds [here](https://docs.travis-ci.com/user/languages/haskell/#specifying-cabal-install-verison).

**Comment by TristanCacqueray** [comment]
Oh my bad, I misread the previous comment :) Thank you for the guidance!

**Comment by TristanCacqueray** [comment]
Ok so running cabal>3 doesn't seem to work with travis as it insists on configuring tests for dependencies, which doesn't seems supported by cabal>3.

Let see if using cabal-2.4 helps, as suggested by https://github.com/kowainik/summoner/blob/master/summoner-cli/examples/cabal-full/.travis.yml

**Comment by TristanCacqueray** [comment]
@CristhianMotoche that PR seems to be ready now, not sure if OSX has been covered though.

File: .travis/ghcup.sh [hunk 1]
```diff
@@ -4,3 +4,8 @@ ghcup install $GHC_VER
 ghcup install-cabal $CABAL_VER
 ghcup set $GHC_VER
 cabal update
+echo "Cabal version before upgrade:"
+cabal --version
+cabal install cabal-install-2.4.0.0
+echo "Cabal version after upgrade:"
+cabal --version
```
Comment (CristhianMotoche) [inline .travis/ghcup.sh:11]: Hi @TristanCacqueray I think we're already installing `cabal-2.4`. We specify the version [here](https://github.com/stackbuilders/dotenv-hs/blob/master/.travis.yml#L14-L16). So, I don't think these changes are necessary.

**Comment by CristhianMotoche** [comment]
Hi @TristanCacqueray The PR look great. I just left you one last comment in your code. After that, I'll approve this PR. Thank you so much for working on it. :smile:

**Comment by TristanCacqueray** [comment]
@CristhianMotoche you're welcome, thank you for adding `good first issue` tag, that was quite instructive!

**Pull request merged**